src/ir/LocalGraph.cpp

```diff
@@ -301,22 +297,120 @@ struct LocalGraph::LocalGraphFlower
     // Bump the current iteration for the next time we are called.
     currentIteration++;
   }
+
+  // When the LocalGraph is in lazy mode we do not compute all of getSetsMap
+  // initially, but instead fill in these data structures that let us do so
+  // later for individual gets. Specifically we need to find the location of a
+  // local.get in the CFG.
+  struct BlockLocation {
+    // The basic block an item is in.
+    FlowBlock* block = nullptr;
+    // The index in that block that the item is at.
+    Index index;
+  };
+  std::unordered_map<LocalGet*, BlockLocation> getLocations;
+
+  // Set up getLocations using the flow blocks, so that we are ready to handle
+  // later lazy requests for the sets of particular gets. This is done in lazy
+  // mode.
+  void prepareLaziness() {
+    prepareFlowBlocks();
+
+    for (auto& block : flowBlocks) {
+      const auto& actions = block.actions;
+      for (Index i = 0; i < actions.size(); i++) {
+        if (auto* get = actions[i]->dynCast<LocalGet>()) {
+          getLocations[get] = BlockLocation{&block, i};
+        }
+      }
+    }
+  }
+
+  // Flow a specific get. This is done in lazy mode.
+  void flowGet(LocalGet* get) {
+    auto index = get->index;
+
+    // Regardless of what we do below, ensure an entry for this get, so that we
+    // know we computed it.
+    auto& sets = getSetsMap[get];
+
+    auto [block, blockIndex] = getLocations[get];
+    if (!block) {
+      // We did not find location info for this get, which means it is
+      // unreachable.
+      return;
+    }
+
+    // We must have the get at that location.
+    assert(blockIndex < block->actions.size());
+    assert(block->actions[blockIndex] == get);
+
+    if (!hasSet[index]) {
+      // As in flow(), when there is no local.set for an index we can just mark
+      // the only writer as the default value.
+      sets.insert(nullptr);
+      return;
+    }
+
+    // Go backwards in this flow block, from the get. If we see other gets that
+    // have not been computed then we can accumulate them as well, as the
+    // results we compute apply to them too.
+    std::vector<LocalGet*> gets;
+    gets.push_back(get);
+    while (blockIndex > 0) {
+      blockIndex--;
```

Comment:
```suggestion
    while (blockIndex-- > 0) {
```
A fantastic opportunity to break out the fabled "goes to" operator, `-->`.

Reply:
I find this slightly harder to read. How useful do you find it?

Reply:
I'm fine not making this change.